src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh

```diff
@@ -25,50 +25,50 @@ fi
 # Shell integration was disabled by the shell, exit without warning assuming either the shell has
 # explicitly disabled shell integration as it's incompatible or it implements the protocol.
 if [ -z "$VSCODE_SHELL_INTEGRATION" ]; then
-	return
+	builtin return
 fi
 
 __vsc_in_command_execution="1"
 __vsc_last_history_id=0
 
 __vsc_prompt_start() {
-	printf "\033]633;A\007"
+	builtin printf "\033]633;A\007"
 }
 
 __vsc_prompt_end() {
-	printf "\033]633;B\007"
+	builtin printf "\033]633;B\007"
 }
 
 __vsc_update_cwd() {
-	printf "\033]633;P;Cwd=%s\007" "$PWD"
+	builtin printf "\033]633;P;Cwd=%s\007" "$PWD"
 }
 
 __vsc_command_output_start() {
-	printf "\033]633;C\007"
+	builtin printf "\033]633;C\007"
 }
 
 __vsc_continuation_start() {
-	printf "\033]633;F\007"
+	builtin printf "\033]633;F\007"
 }
 
 __vsc_continuation_end() {
-	printf "\033]633;G\007"
+	builtin printf "\033]633;G\007"
 }
 
 __vsc_right_prompt_start() {
-	printf "\033]633;H\007"
+	builtin printf "\033]633;H\007"
 }
 
 __vsc_right_prompt_end() {
-	printf "\033]633;I\007"
+	builtin printf "\033]633;I\007"
 }
 
 __vsc_command_complete() {
-	local __vsc_history_id=$(history | tail -n1 | awk '{print $1;}')
+	builtin local __vsc_history_id=$(builtin history | tail -n1 | awk '{print $1;}')
```

Comment:
Same, is builtin needed here?

Reply:
kitty does that, we should too

Reply:
https://github.com/kovidgoyal/kitty/blob/master/shell-integration/zsh/kitty-integration#L66